### PR TITLE
NO-REF unified react webpack

### PR DESCRIPTION
**Pull request by hokei**
[Project Reno Jira Ticket](http://jira.nypl.org/browse/RENO-XXX)

[ResearchNow Jira Ticket](http://jira.nypl.org/browse/SFR-XXX)

## **This PR does the following:**
- Allows design-system-react-components to be released with styles, eliminating the need to pull design-system-styles separately

### Front End Review:
- [ ] View [the example in Storybook](https://reno-XXX-nypl.pantheonsite.io/themes/custom/nypl_emulsify/pattern-lab/public)
- [ ] Check against the [Metronome documentation](http://themetronome.co/components/xxx/?fresh=true)

### CODE REVIEW

File: src/react-components/package.json [hunk 1]
```diff
@@ -21,7 +21,10 @@
   },
   "keywords": [],
   "dependencies": {
-    "@nypl/design-system-icons": "file:../icons"
+    "@nypl/design-system-icons": "file:../icons",
+    "@nypl/design-system-styles": "file:../styles",
+    "mini-css-extract-plugin": "^0.9.0",
```
Comment (EdwinGuzman) [inline src/react-components/package.json]: This can be a devdep, as well as sass-loader.

File: storybook/package.json [hunk 2]
```diff
@@ -36,5 +37,8 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
+    "webpack-cli": "^3.3.11"
```
Comment (EdwinGuzman) [inline storybook/package.json]: Why do you have to install this for multiple folders?

Comment (hokei) [reply storybook/package.json]: aaahhh I did it through Lerna and didn't hoist.  Wish it would ask you which folder you needed it in by default ...

**Comment by EdwinGuzman** [comment]
This works well when testing it (through `npm link`) on a SimplyE app. I would just say the README should be updated to reflect that the design-system-styles package is no longer needed. And now, all you need is `@import "~@nypl/design-system-react-components/dist/styles.scss";` in the app's main scss file.

File: src/twig/package.json [hunk 3]
```diff
@@ -14,9 +14,14 @@
...
-    "puppeteer": "^2.0.0"
+    "puppeteer": "^2.0.0",
+    "webpack-cli": "^3.3.11"
   },
   "publishConfig": {
     "access": "public"
+  },
+  "dependencies": {
+    "extract-text-webpack-plugin": "^3.0.2",
+    "sass-loader": "^8.0.2"
```
Comment (EdwinGuzman) [inline src/twig/package.json]: This PR is for the react components package, so are these updates necessary? And if so, what are they doing? Or is this similar to the Lerna hoisting you mentioned in a different comment?

**Comment by EdwinGuzman** [comment]
Update looks good... but tests :(

**Comment by hokei** [comment]
yeah still figuring out why it's not finding `node_modules/.staging` files.  I can't repro reliably on my machine ...

**Review by EdwinGuzman — APPROVED** [review]

**Pull request merged**